modules/tinymce/src/themes/silver/main/ts/Events.ts

```diff
@@ -6,12 +6,15 @@
  */
 
 import { Attachment, Channels, Gui, SystemEvents } from '@ephox/alloy';
-import { document, MouseEvent, Node as DomNode, UIEvent } from '@ephox/dom-globals';
+import { MouseEvent, Node as DomNode, UIEvent } from '@ephox/dom-globals';
 import { Arr } from '@ephox/katamari';
-import { DomEvent, Element, EventArgs } from '@ephox/sugar';
+import { DomEvent, Element, EventArgs, ShadowDom } from '@ephox/sugar';
 import Editor from 'tinymce/core/api/Editor';
 
 const setup = (editor: Editor, mothership: Gui.GuiSystem, uiMothership: Gui.GuiSystem) => {
+
+  const root = ShadowDom.getRootNode(Element.fromDom(editor.getElement()));
```

Comment:
Just a heads up, this will break things, as it **must** have a listener on the document, otherwise when you click outside the editor it won't close the menus. As an example, with the current changes it'll only close if you click somewhere inside the shadow dom.

As such this likely needs to be a new set of listeners, so we have one for the document and one for the shadow root (if inside a shadow root).

Reply:
Done.

Reply:
Actually, doing that broke clicking on the format menu. I've reverted it for now.

Reply:
Looking at https://developers.google.com/web/fundamentals/web-components/shadowdom#events, it says most events bubble up out of a shadow dom anyways, so I have to wonder if this change is even needed?

It could also be the fact that it retargets the event target when bubbling up that's causing an issue maybe?

Reply:
Without this change, menus don't work. With the change, menus do work. Need to look into it further.